### PR TITLE
FS 1510 email application confirmation to notification bug fix

### PR DESCRIPTION
**Pull request by ramsharma-prog**
Bug fix when posting application to Notification service

### CODE REVIEW

**Review by SentientHousePlant — CHANGES_REQUESTED** [review]

File: external_services/http_methods.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
     response = requests.post(endpoint, json=params)
     if response.status_code in [200, 201]:
+        current_app.logger.info(
+            "Post successfully sent to {endpoint} with response code:"
+            f" '{response.status_code}'."
+        )
+
         return response.json()
+
+    raise CustomError(
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply external_services/http_methods.py]: Updated 😄

File: .github/workflows/manual-dev-deploy.yaml [hunk 2]
```diff
@@ -30,3 +30,12 @@
             cf_username: ${{secrets.CF_USER}}
             cf_password: ${{secrets.CF_PASSWORD}}
             command: push funding-service-design-application-store-dev
+        - name: Apply network policy for notification service
```
Comment (Adam-W1) [inline .github/workflows/manual-dev-deploy.yaml:33]: do we still need this?

File: external_services/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class NotificationError(Exception):
+    """Exception raises an an error
+
+    Attributes:
+        message -- explanation of the error
+    """
+
+    def __init__(self, message="Sorry, there was a problem please try later"):
```
Comment (Adam-W1) [inline external_services/exceptions.py:8]: All for more accurate errors :) should we add 'problem to the notification service' here?

File: external_services/http_methods.py [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
         params = {k: v for k, v in params.items() if v is not None}
     current_app.logger.info(
         f"Attempting POST to the following endpoint: '{endpoint}'."
     )
     response = requests.post(endpoint, json=params)
     if response.status_code in [200, 201]:
+        current_app.logger.info(
+            "Post successfully sent to {endpoint} with response code:"
+            f" '{response.status_code}'."
+        )
```
Comment (Adam-W1) [inline external_services/http_methods.py:18]: 👌

**Review by Adam-W1 — CHANGES_REQUESTED** [review]
just a couple small comments :)

File: .github/workflows/manual-dev-deploy.yaml (hunk 2 above)
Comment (ramsharma-prog) [reply .github/workflows/manual-dev-deploy.yaml:33]: Throws an error if removed so probably yes

File: external_services/exceptions.py (hunk 3 above)
Comment (ramsharma-prog) [reply external_services/exceptions.py]: Error updated!!

**Review by SentientHousePlant — APPROVED** [review]

**Review by harryyo — APPROVED** [review]

**Pull request merged**